### DISCUSSION TITLE
how to setup now for next.js using routes

### DISCUSSION THREAD

**Discussion by chrisdrackett** [vercel/vercel#3938] [General] [answered]
I have a next.js site that I'm deploying to now and at the moment it works for the homepage. The problem is if I go to, say `/blog/123` in the browser I still get the index page. This all works locally so I assume its something in the way I have `now.json` setup:

```
  "builds": [
    {
      "src": "web/package.json",
      "use": "@now/next"
    },
    {
      "src": "server/src/index.ts",
      "use": "@now/node",
      "config": { "maxLambdaSize": "50mb" }
    }
  ],
  "routes": [
    {
      "src": "/api",
      "status": 301,
      "headers": { "Location": "/api/" }
    },
    { "src": "/api/.*", "dest": "server/src/index.ts" },
    {
      "src": "/_next/(.*)",
      "dest": "/web/_next/$1"
    },
    {
      "src": "/.*",
      "dest": "web/index"
    }
  ],
```

I'm guessing something is wrong with either my `src` or `dest` in that final route, but I can't seem to find any related docs as I'm using an older format for my now.json

**Comment by chrisdrackett** [comment]
seems to me the following works, not 100% sure why the `continue` is required, but it breaks without:

```
"builds": [
    {
      "src": "web/package.json",
      "use": "@now/next"
    },
    {
      "src": "server/src/index.ts",
      "use": "@now/node",
      "config": { "maxLambdaSize": "50mb" }
    }
  ],
  "routes": [
    {
      "src": "/api",
      "status": 301,
      "headers": { "Location": "/api/" }
    },
    { "src": "/api/.*", "dest": "server/src/index.ts" },
    {
      "src": "(.*)",
      "dest": "/web$1",
      "continue": true
    }
  ],
```

**Reply by paulogdm** [reply]
Any reason for not using Next.js API functionality?

**Reply by chrisdrackett** [reply]
we abandoned doing anything server related on Next.js because we had too many issues with database connections maxing out, connection pooling causing errors, having to move to a more expensive database host to maybe fix the issue. We just decided to move our API to Heroku directly and now only have our Next.js app on now. I really love the idea of the API functionality, but when it comes to our app and database needs it just didn't seem viable for production.

**Comment by paulogdm** [comment, marked as answer]
This approach is not recommended. Using the [Next.js API functionality](https://nextjs.org/docs/api-routes/introduction) directly unlocks a few features and also allows you to drop "builds" in favor of a cleaner approach. Consider using it in the future.
